### ISSUE TITLE
Extracting internal simple demangler as a executable file

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#296]
Sometimes, it's difficult to understand internal simple demangler's behavior. I think if there's an external tool such as `demangler` as an executable, that would be really helpful to understand and test it in a long term.

**Comment by namhyung** [comment]
OK, it's fairly self-contained and should be easy to separate it.

**Comment by honggyukim** [comment]
Thanks. Please do that when you have free time.

**Comment by honggyukim** [comment]
Hi @namhyung, I've implemented basic version of demangler as discussed in #310.
I didn't remove any extra dependency yet but just made it very simple to use.
Please let me know how you think about it.

**Comment by honggyukim** [comment]
The basic usages is as follows:
```
$ ./demangler _ZNSt5tupleIJPbSt14default_deleteIA_bEEEC2Ev 
std::tuple::tuple

$ ./demangler --simple _ZNSt5tupleIJPbSt14default_deleteIA_bEEEC2Ev 
std::tuple::tuple

$ ./demangler --full _ZNSt5tupleIJPbSt14default_deleteIA_bEEEC2Ev 
std::tuple<bool*, std::default_delete<bool []> >::tuple()
```

**Comment by namhyung** [comment]
Closed by #310.
